libcodechecker/analyze/plist_parser.py

```diff
@@ -433,19 +433,32 @@ def __write_bugs(self, output, reports, files, analyzed_source_file,
         report_stats["files"] = file_stats
         report_stats["reports"] = report_count
 
-    def parse_and_write(self, plist_file, analyzed_source_file,
-                        output=sys.stdout):
+    @staticmethod
+    def parse(plist_file):
         """
-        Parse a plist report file format it to a more human readable format.
+        Parse a plist report file.
+        Returns:
+            - list of source files
+            - list of reports (type Report)
         """
-        report_stats = {}
+        files, reports = [], []
         try:
             files, reports = parse_plist(plist_file)
         except Exception as ex:
             traceback.print_stack()
             LOG.error('The generated plist is not valid!')
             LOG.error(ex)
-            return 1
+        finally:
+            return files, reports
+
+    def write(self, files, reports, analyzed_source_file, output=sys.stdout):
+        """
+        Format an already parsed plist report file to a more
+        human readable format.
+        The formatted text is written to the output.
+        During writing the output statistics are collected.
+        """
+        report_stats = {}
 
         self.__write_bugs(output, reports, files, analyzed_source_file,
```

Comment:
I wonder if it still makes sense to have the `__write_bugs` method, or we could just inline it here.

Reply:
done